File: 4_Dicas/4.4_Interfaces_gráficas/Ex21.c
#include <gtk/gtk.h>

void end_program(GtkWidget *wid, gpointer ptr)
{
	gtk_main_quit();
}

static void col_selected(GtkColorButton *btn, gpointer ptr)
{
	// Mostrar no terminal a cor selecionada
	GdkColor col;
	gtk_color_button_get_color(btn, &col);
	printf("red = %d; green = %d; blue = %d\n", col.red, col.green, col.blue);
}

void main(int argc, char *argv[])
{
	gtk_init(&argc, &argv);
	GtkWidget *win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
	GtkWidget *btn = gtk_button_new_with_label("Close window");
	g_signal_connect(btn, "clicked", G_CALLBACK(end_program), NULL);
	g_signal_connect(win, "delete_event", G_CALLBACK(end_program), NULL);
	GtkWidget *vbox = gtk_vbox_new(FALSE, 5);
	gtk_container_add(GTK_CONTAINER(win), vbox);
	// Adicionar botão de seleção de cor
	GtkWidget *col_btn = gtk_color_button_new();
	g_signal_connect(col_btn, "color-set", G_CALLBACK(col_selected), NULL);
	gtk_box_pack_start(GTK_BOX(vbox), col_btn, TRUE, TRUE, 0);
	gtk_box_pack_start(GTK_BOX(vbox), btn, TRUE, TRUE, 0);
	gtk_widget_show_all(win);
	gtk_main();
}

File: 4_Dicas/4.4_Interfaces_gráficas/Python/Ex21.py
# https://python-gtk-3-tutorial.readthedocs.io/pt_BR/latest/introduction.html
import gi, sys
gi.require_version("Gtk", "3.0")
from gi.repository import Gtk

class MyWindow(Gtk.Window):
	def __init__(self):
		super().__init__(title=sys.argv[0])
		self.btn = Gtk.Button(label="Close window")
		self.btn.connect("clicked", Gtk.main_quit)
		self.col_btn = Gtk.ColorButton()
		self.col_btn.connect("color-set", self.col_selected)
		self.tbl = Gtk.Grid()
		self.tbl.set_column_homogeneous(True)
		self.tbl.add(self.col_btn)
		self.tbl.attach(self.btn, 0, 1, 1, 1)
		self.add(self.tbl)
		self.connect("destroy", Gtk.main_quit)
		self.show_all()
		Gtk.main()
	def col_selected(self, widget):
		col = self.col_btn.get_rgba()
		print("red = %d; green = %d; blue = %d" % (col.red*65535, col.green*65535, col.blue*65535))

win = MyWindow()